src/netspresso_trainer/pipelines/detection.py

```diff
@@ -102,18 +117,77 @@ def test_step(self, batch):
         return results
 
     def get_metric_with_all_outputs(self, outputs):
-        targets = np.empty((0, 4))
-        preds = np.empty((0, 5))  # with confidence score
-        targets_indices = np.empty(0)
-        preds_indices = np.empty(0)
+        pred = []
+        targets = []
         for output_batch in outputs:
             for detection, class_idx in output_batch['target']:
-                targets = np.vstack([targets, detection])
-                targets_indices = np.append(targets_indices, class_idx)
+                target_on_image = {}
+                target_on_image['boxes'] = detection
+                target_on_image['labels'] = class_idx
+                targets.append(target_on_image)
 
             for detection, class_idx in output_batch['pred']:
-                preds = np.vstack([preds, detection])
-                preds_indices = np.append(preds_indices, class_idx)
-
-        pred_bbox, pred_confidence = preds[..., :4], preds[..., -1]  # (N x 4), (N,)
-        self.metric_factory.calc((pred_bbox, preds_indices, pred_confidence), (targets, targets_indices), phase='valid')
+                pred_on_image = {}
+                pred_on_image['post_boxes'] = detection[..., :4]
+                pred_on_image['post_scores'] = detection[..., -1]
+                pred_on_image['post_labels'] = class_idx
+                pred.append(pred_on_image)
+        self.metric_factory(pred, target=targets, phase='valid')
+        
+    def save_checkpoint(self, epoch: int):
```

Comment:
@illian01 

If we override task-specific `save_checkpoint` function, it may be better to express as `abstractmethod` in base pipeline.
Someone may be hard to find where it is defined..

Reply:
@deepkyu 
I overrided `save_checkpoint` method because we have to separate backbone and head for detection task.
Thus, this overrided method don't need if we serve fx model with integrated one.
I think it would be better to remove this method when we can compress and benchmark the integrated detection task model.
I will fix `save_checkpoint` to `abstractedmethod` if we determined to serve detection model as is (backbone and head separately) through internal discussion.